plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -840,15 +1025,24 @@ public void rollbackCreateTable(ConnectorSession session, JdbcOutputTableHandle
     }
 
     @Override
-    public String buildInsertSql(JdbcOutputTableHandle handle, List<WriteFunction> columnWriters)
+    public boolean includePageSinkIdColumn(ConnectorSession session)
+    {
+        return supportsFTE && !isNonTransactionalInsert(session);
```

Comment:
I am not super convinced we should be using the new protocol if retries are not enabled, even if we do support it.
But I will not force the other way.